lib/ansible/modules/cloud/vmware/vmware_vcenter_permission.py

```diff
@@ -0,0 +1,359 @@
+---
+module: vmware_vcenter_permission
+short_description: manages permissions on vCenter
+description:
+    - creates, updates or deletes permissions on vCenter objects
+version_added: "2.9"
+author: Christian Neugum (@digifuchsi)
+options:
+    entity:
+        description:
+            - describes vCenter object to apply permission on
+        required: true
+        type: dict
+        options:
+            name:
+                description:
+                    - name of object
+                required: true
+                type: str
+            folder:
+                description:
+                    - folder of object
+                required: true
+                type: str
+            type:
+                description:
+                    - type of object
+                required: true
+                type: str
+                choices: ['vCenter', 'Datacenter', 'Cluster', 'ResourcePool',
+                          'Host', 'Folder', 'DistributedSwitch',
+                          'DistributedPortgroup', 'NsxtNetwork']
+    principal:
+        description:
+            - describes user to set permission to
+        required: true
+        type: dict
+        options:
+            name:
+                description:
+                    - name of user
+                required: true
+                type: str
+            domain:
+                description:
+                    - domain of user
+                required: true
+                type: str
+            isGroup:
+                description:
+                    - indicates if user is a group
+                default: false
+                type: bool
+    role_name:
+        description:
+            - Name of the role to assign
+            - required if state is present
+        required: false
+        type: str
+    propagate:
+        description:
+            - set to true if permisison should be propagated to children
+        required: false
+        type: bool
+        default: false
+    state:
+        description:
+            - set to present to add/update permisisons, absent to delete them
+        required: false
+        type: str
+        choices: ['present', 'absent']
+        default: 'present'
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: add or update permission on datacenter DC
+  vmware_vcenter_permission:
+    hostname: "{{ vCenter_hostname }}"
+    username: "{{ vCenter_username }}"
+    password: "{{ vCenter_password }}"
+    validate_certs: False
+    role_name: 'VM-Power'
+    entity:
+          name: 'DC'
+          type: 'Datacenter'
+    principal:
+          name: 'test'
+          domain: 'vsphere.local'
+          isGroup: False
+    propagate: True
+    state: present
+
+- name: remove permission from datacenter DC
+  vmware_vcenter_permission:
+    hostname: "{{ vCenter_hostname }}"
+    username: "{{ vCenter_username }}"
+    password: "{{ vCenter_password }}"
+    validate_certs: False
```

Comment:
The users should validate the certificate. I would prefer you remove the `validate_certs: False` from the example, this  way the official documentation won't come with bad practice.